Source/Core/CameraLimiter.js

```diff
@@ -0,0 +1,301 @@
+     */
+    CameraLimiter.prototype.closestOrientationTo = function(orientation) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('orientation', orientation);
+        if (!(orientation instanceof HeadingPitchRoll)) {
+            throw new DeveloperError('orientation required to be of type HeadingPitchRoll.');
+        }
+
+        // if min is defined, max must also be. if min is not defined, max must also not be.
+        if (!defined(this.minHeadingPitchRoll) && defined(this.maxHeadingPitchRoll)) {
+            throw new DeveloperError('minHeadingPitchRoll is required.');
+        }
+        if (defined(this.minHeadingPitchRoll) && !defined(this.maxHeadingPitchRoll)) {
+            throw new DeveloperError('maxHeadingPitchRoll is required.');
+        }
+        //>>includeEnd('debug');
+
+        if(!defined(this.minHeadingPitchRoll)) {
+            return orientation;
+        }
+
+        var returningOrientation = orientation;
+
+        returningOrientation.heading = Math.max(this.minHeadingPitchRoll.heading, Math.min(this.maxHeadingPitchRoll.heading, returningOrientation.heading));
+        returningOrientation.pitch = Math.max(this.minHeadingPitchRoll.pitch, Math.min(this.maxHeadingPitchRoll.pitch, returningOrientation.pitch));
+        returningOrientation.roll = Math.max(this.minHeadingPitchRoll.roll, Math.min(this.maxHeadingPitchRoll.roll, returningOrientation.roll));
+
+        return returningOrientation;
+    };
+
+    /**
+     * Duplicates this CameraLimiter instance.
+     *
+     * @param {CameraLimiter} [result] The object onto which to store the result.
+     * @returns {CameraLimiter} The modified result parameter or a new CameraLimiter instance if one was not provided.
+     */
+    CameraLimiter.clone = function(limiter, result) {
+        if (!defined(result)) {
+            var options = {};
+            result = new CameraLimiter(options);
+        }
+
+        if (!defined(limiter.boundingObject)) {
+            result.boundingObject = undefined;
+        } else if (limiter.boundingObject instanceof AxisAlignedBoundingBox) {
+            result.boundingObject = AxisAlignedBoundingBox.clone(limiter.boundingObject, result.boundingObject);
+        } else if (limiter.boundingObject instanceof BoundingRectangle) {
+            result.boundingObject = BoundingRectangle.clone(limiter.boundingObject, result.boundingObject);
+        } else if (limiter.boundingObject instanceof BoundingSphere) {
+            result.boundingObject = BoundingSphere.clone(limiter.boundingObject, result.boundingObject);
+        } else if (limiter.boundingObject instanceof OrientedBoundingBox) {
+            result.boundingObject = OrientedBoundingBox.clone(limiter.boundingObject, result.boundingObject);
+        } else {
+            //>>includeStart('debug', pragmas.debug);
+            throw new DeveloperError('bounding object is not of an allowed type');
+            //>>includeEnd('debug');
+        }
```

Comment:
I think these all have prototype clone methods. This can be shortened to

```
if (defined(limiter.boundingObject)) {
    limiter.boundingObject.clone(result);
}
```

and there is no need to throw the developer error.